src/libraries/drand/_index.md

```diff
@@ -0,0 +1,120 @@
+---
+menuTitle: drand
+title: drand - Distributed Randomness
+---
+
+DRand (Distributed Randomness) is a publicly verifiable random beacon protocol Filecoin relies on as a source of unbiasable entropy for leader election (see {{<sref leader_election>}}).
+
+At a high-level, the drand protocol runs a series of MPCs (Multi-Party Computations) in order to produce a series of deterministic, verifiable random values. Specifically, after a trusted setup, a known (to each other) group of n drand nodes sign a given message using t-of-n threshold BLS signatures in a series of successive rounds occuring at regular intervals (the drand round time).
+Any node that has gathered t of the signatures can reconstruct the full BLS signature. This signature is then hashed using SHA-512 in order to produce a collective random value which can be verified against the collective public key generated during the trusted setup.
+
+drand assumes that at least t of the n nodes are honest (and online -- for liveness). If this threshold is broken, the adversary can permanently halt randomness production but cannot otherwise bias the randomness.
+
+You can learn more about how drand works, by visiting its [repository](https://github.com/drand/drand), or reading its [spec](https://github.com/drand/drand/blob/master/docs/SPECS.md).
+
+In the following sections we look in turn at how the Filecoin protocol makes use of drand randomness, and at some of the characteristics of the specific drand network Filecoin uses.
+
+### Drand randomness outputs
+
+By polling the appropriate endpoint (see below for specifics on the drand network Filecoin uses), a Filecoin node will get back a drand value formatted as follows (e.g.):
+
+```
+{
+  "round": 367,
+  "signature": "b62dd642e939191af1f9e15bef0f0b0e9562a5f570a12a231864afe468377e2a6424a92ccfc34ef1471cbd58c37c6b020cf75ce9446d2aa1252a090250b2b1441f8a2a0d22208dcc09332eaa0143c4a508be13de63978dbed273e3b9813130d5",
+  "previous_signature": "afc545efb57f591dbdf833c339b3369f569566a93e49578db46b6586299422483b7a2d595814046e2847494b401650a0050981e716e531b6f4b620909c2bf1476fd82cf788a110becbc77e55746a7cccd47fb171e8ae2eea2a22fcc6a512486d",
+  "randomness": "d7aed3686bf2be657e6d38c20999831308ee6244b68c8825676db580e7e3bec6"
+}
+```
+
+Specifically, we have:
+
+- `Signature`           -- the threshold BLS signature on the previous signature value `Previous` and the current round number `round`.
+- `PreviousSignature`   -- the threshold BLS signature from the previous drand round.
+- `Randomness`          --  the SHA256 hash of `Signature`, to be used as the random value for this round.
+- `Round`               -- the index of Randomness in the sequence of all random values produced by this drand network.
+
+Specifically, the message signed is the concatenation of the round number treated as a uint64 and the previous signature. At the moment, drand uses BLS signatures on the BN256 curves and the signature is made over G1.
```

Comment:
drand uses the BLS12-381 curve with the latest v7 RFC of hash-to-curve - link to [spec](https://github.com/drand/drand/blob/master/docs/SPECS.md#cryptographic-specification)

Reply:
I see. Will fix. Current drand readme is stale then fyi